secboot/secboot_sb.go

```diff
@@ -90,13 +90,13 @@ func UnlockVolumeUsingSealedKeyIfEncrypted(disk disks.Disk, name string, sealedE
 		}
 		// otherwise it is an error not found and we should search for the
 		// unencrypted device
-		partUUID, err = disk.FindMatchingPartitionUUIDWithFsLabel(name)
+		part, err = disk.FindMatchingPartitionWithFsLabel(name)
 		if err != nil {
 			return res, fmt.Errorf("error enumerating partitions for disk to find unencrypted device %q: %v", name, err)
 		}
 	}
 
-	partDevice := filepath.Join("/dev/disk/by-partuuid", partUUID)
+	partDevice := filepath.Join("/dev/disk/by-partuuid", part.PartitionUUID)
```

Comment:
is it intentional to keep using by-partuuid here?

Reply:
That requires more changes in snap-bootstrap. For now it is not needed. The main point here is to simplify everywhere where we need to get keys from keyring.

We will probably have to refactor snap-bootstrap to match primary keys, and we can do that refactoring in the same time.